rotkehlchen/db/filtering.py

```diff
@@ -744,3 +770,14 @@ def make(
         filters.append(filter_query.timestamp_filter)
         filter_query.filters = filters
         return filter_query
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class DBProtocolsFilter(DBFilter):
+    protocols: List[str]
+
+    def prepare(self) -> Tuple[List[str], List[Any]]:
```

Comment:
This will fail for empty list of protocols. But if we stop empty list of protocols at the api level it should be impossible.

So please write a comment here specifying that `protocols` can't be empty since it's checked by flask (implement said check and test it)

Reply:
Added check, test and a comment